util.py

```diff
@@ -260,14 +260,22 @@ def create_executable(filename, command):
     )
 
 
-def check_file_sha256_checksum(filename, reference_checksum):
+def compute_file_sha256_checksum(filename):
     """
     Computes the sha256sum for a file
 
     :param str filename: a single file to be checked
-    :param str reference_checksum: reference sha256 checksum
-    :return:
+    :return: checksum
+    :rtype:str
     """
     checksum_command_output = sp.check_output(["sha256sum", filename])
-    file_checksum = checksum_command_output.decode().strip().split(" ")[0]
-    assert file_checksum == reference_checksum
+    return checksum_command_output.decode().strip().split(" ")[0]
+
+
+def check_file_sha256_checksum(filename, reference_checksum):
+    """
+    Validates the sha256sum for a file against the target checksum
+
+    :param str filename: a single file to be checked
+    """
+    assert compute_file_sha256_checksum(filename) == reference_checksum
```

Comment:
Since this is only one assert, maybe it would be easier to have this in the tests themselves? Because I imagine right now if this assert fails the error will be given in the util file and then you have to search the trace for the call. This might make debugging / finding out what failed more tedious. 
If this logic is extended / larger then I agree with sourcing it out.

Reply:
As the testing is run with `pytest`, it will directly show the name of the tests that are failing. But yes, the assert could also be part of the test, maybe this is better style.

Reply:
Ahh okay. If the unit shows which test fails its not that big of a deal, then its just personal preference/style. Then we should not over think this. Just decide on something, both are fine I guess.

Reply:
pytest also tells you the line of the failing assert